### PR TITLE
test_aux_data: rewrite test cases in criterion

### LINKED ISSUE #2579: convert ut test_aux_data to criterion
Use Criterion to rewrite this unit test.

### PR DESCRIPTION
**Pull request by mehul-m-prajapati**
fixes #2579

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
do nothing -> CI won't start)

**Comment by Kokan** [comment]
@kira-syslogng ok to test

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/transport/tests/test_aux_data.c [hunk 1]
```diff
@@ -20,7 +20,8 @@
  * OpenSSL libraries as published by the OpenSSL project. See the file
  * COPYING for details.
  */
-#include "testutils.h"
+#include <criterion/criterion.h>
+#include "apphook.h"
 #include "transport/transport-aux-data.h"
 
 #define AUX_DATA_TESTCASE(x, ...) do { aux_data_testcase_begin(#x, #__VA_ARGS__); x(__VA_ARGS__); aux_data_testcase_end(); } while(0)
```
Comment (Kokan) [inline lib/transport/tests/test_aux_data.c]: Could you please remove this unused macros ?

File: lib/transport/tests/test_aux_data.c [hunk 2]
```diff
@@ -117,8 +116,7 @@ test_aux_data_added_nvpairs_are_returned_by_foreach_in_order(void)
   free_aux(aux);
 }
 
-static void
-test_aux_data_copy_creates_an_identical_copy(void)
+Test(aux_data, test_aux_data_copy_creates_an_identical_copy)
 {
   LogTransportAuxData *aux = construct_aux_with_some_data();
```
Comment (furiel) [inline lib/transport/tests/test_aux_data.c]:
```
LogTransportAuxData *aux = construct_aux_with_some_data();
[...]
free_aux(aux);
```
are repetitive pattern. Aux could be constructed in the `TestSuite` setup function, free could be called in the teardown function.

Comment (mehul-m-prajapati) [reply lib/transport/tests/test_aux_data.c]: Yes, I will change it.

File: lib/transport/tests/test_aux_data.c (hunk 1 above)
Comment (mehul-m-prajapati) [reply lib/transport/tests/test_aux_data.c]: Ok. I will remove those macros.

Comment (mehul-m-prajapati) [reply lib/transport/tests/test_aux_data.c]: ok

File: lib/transport/tests/test_aux_data.c (hunk 2 above)
Comment (mehul-m-prajapati) [reply lib/transport/tests/test_aux_data.c]: Yes, that is a good suggestion.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by MrAnno** [comment]
Thanks!

**Pull request merged**